doc/api/stream.md

```diff
@@ -61,6 +61,15 @@ Stream instances are switched into object mode using the `objectMode` option
 when the stream is created. Attempting to switch an existing stream into
 object mode is not safe.
 
+The object mode indicates that the stream accepts and/or emits values which are
+not limited to `Buffer` or strings. Be aware that streams which are in object
+mode can be piped to streams not in the object mode but an exception will
+be raised if you attempt to write something except a `Buffer` or a string. To
+prevent this create an appropriate implementation to
+[transform][stream-_transform] the values into buffers or strings. The stream
+which transforms the values has to be in the object mode as well to be able to
+accept objects.
```

Comment:
Note that you can specify only one side in object mode, so it correctly buffers based on the datatype.